geode-core/src/main/java/org/apache/geode/internal/cache/BucketRegion.java

```diff
@@ -557,6 +557,34 @@ public boolean virtualPut(EntryEventImpl event, boolean ifNew, boolean ifOld,
     }
   }
 
+  @Override
+  void cmnClearRegion(RegionEventImpl regionEvent, boolean cacheWrite, boolean useRVV) {
+    if (getBucketAdvisor().isPrimary()) {
+      logger.info("Not primary bucket when doing clear, do nothing");
+      return;
+    }
+
+    boolean enableRVV = useRVV && getConcurrencyChecksEnabled();
+    RegionVersionVector rvv = null;
+    if (enableRVV) {
+      rvv = getVersionVector();
+    }
+
+    // get rvvLock
+    Set<InternalDistributedMember> participants =
+        getCacheDistributionAdvisor().adviseInvalidateRegion();
+    try {
+      obtainWriteLocksForClear(regionEvent, participants);
+      clearRegionLocally(regionEvent, cacheWrite, null);
+      if (!regionEvent.isOriginRemote() && regionEvent.getOperation().isDistributed()) {
```

Comment:
Shouldn't it only be the origin that distributes?

Reply:
Yes, after I carefully reviewed the code, we don't need this check.